serializers/src/glTF/2.0/babylon.glTFMaterial.ts

```diff
@@ -370,6 +383,40 @@ module BABYLON.GLTF2 {
             glTFMaterial.pbrMetallicRoughness = glTFPbrMetallicRoughness;
 
             materials.push(glTFMaterial);
+
+            return Promise.all(promises).then(() => { /* do nothing */ });
+        }
+
+        /**
+         * 
+         * @param texture Texture with alpha to overwrite to one
+         * @param useAlpha Specifies if alpha should be preserved or not
+         * @returns Promise with texture
+         */
+        public static _SetAlphaToOneAsync(texture: BaseTexture, useAlpha: boolean): Promise<Texture> {
+            return Promise.resolve().then((() => {
+                if (useAlpha) {
+                    return Promise.resolve(texture as Texture);
+                }
+                else {
+                    const scene = texture.getScene();
+                    if (scene == null) {
+                        return Promise.reject(`Scene not available for texture ${texture.name}`);
+                    }
+                    else {
+                        const proceduralTexture = new ProceduralTexture('texture', texture.getSize(), 'setAlphaToOne', scene);
+                        if (proceduralTexture == null) {
+                            return Promise.reject(`Cannot create procedural texture for ${texture.name}!`);
+                        }
+                        else {
+                            proceduralTexture.setTexture('textureSampler', texture as Texture);
+                            return scene.whenReadyAsync().then(() => {

```

Comment:
@bghgary is there a way to know when a procedural texture has completed processing?  I only used this line since it was one way to know.   Otherwise I would get black textures on export.

Reply:
You should be able to use onLoadedObservable.